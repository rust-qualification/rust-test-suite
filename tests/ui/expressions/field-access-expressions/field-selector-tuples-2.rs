// Field Access Expressions: 7fc7d0a9-9066-4c99-81a5-37bd9ca6b223

fn main() {
    let my_tuple = (3, "hi", 1.2);
    let _ = my_tuple.first;
    //~^ ERROR no field `first` on type `({integer}, &str, {float})`
    let _ = my_tuple.3;
    //~^ ERROR no field `3` on type `({integer}, &str, {float})`
    let empty_tuple = ();
    let _ = empty_tuple.0;
    //~^ ERROR no field `0` on type `()`
}
