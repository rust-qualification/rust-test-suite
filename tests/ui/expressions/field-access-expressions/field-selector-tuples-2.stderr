error[E0609]: no field `first` on type `({integer}, &str, {float})`
  --> $DIR/field-selector-tuples-2.rs:5:22
   |
LL |     let _ = my_tuple.first;
   |                      ^^^^^ unknown field

error[E0609]: no field `3` on type `({integer}, &str, {float})`
  --> $DIR/field-selector-tuples-2.rs:7:22
   |
LL |     let _ = my_tuple.3;
   |                      ^ unknown field

error[E0609]: no field `0` on type `()`
  --> $DIR/field-selector-tuples-2.rs:10:25
   |
LL |     let _ = empty_tuple.0;
   |                         ^ unknown field

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0609`.
